pkg/cluster/resources.go

```diff
@@ -119,13 +121,73 @@ func (c *Cluster) createStatefulSet() (*v1beta1.StatefulSet, error) {
 	return statefulSet, nil
 }
 
+func getPodIndex(podName string) (int32, error) {
+	parts := strings.Split(podName, "-")
+	if len(parts) == 0 {
+		return 0, fmt.Errorf("pod has no index part")
+	}
+
+	postfix := parts[len(parts)-1]
+	res, err := strconv.ParseInt(postfix, 10, 32)
+	if err != nil {
+		return 0, fmt.Errorf("couldn not parse pod index: %v", err)
+	}
+
+	return int32(res), nil
+}
+
+func (c *Cluster) preScaleDown(newStatefulSet *v1beta1.StatefulSet) error {
+	masterPod, err := c.getRolePods(Master)
+	if err != nil {
+		return fmt.Errorf("could not get master pod: %v", err)
+	}
+
+	podNum, err := getPodIndex(masterPod[0].Name)
+	if err != nil {
+		return fmt.Errorf("could not get pod number: %v", err)
+	}
+
+	//Check if scale down affects current master pod
+	if *newStatefulSet.Spec.Replicas >= podNum+1 {
+		return nil
+	}
+
+	podName := fmt.Sprintf("%s-0", c.Statefulset.Name)
+	masterCandidatePod, err := c.KubeClient.Pods(c.OpConfig.Namespace).Get(podName, metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("could not get master candidate pod: %v", err)
+	}
+
+	if c.podSpiloRole(masterCandidatePod) == Master {
```

Comment:
I'd remove that. It won't happen because of line 151, since we cannot scale down to less than one pod. It might be that the masterCandidatePod will switch to master after line 151 but before line 161, but we cannot guarantee the same won't happen after line 161, and Failover will catch this anyway.